enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py

```diff
@@ -94,53 +99,48 @@ def _manager_on_boot(self):
     def _process_work_order_sync(self, process_wo_id):
         """
         Process the work-order of the specified work-order id
-        and return the response.
-        Used for synchronous execution.
+        and return the response. Used for synchronous execution.
+
         Parameters:
-        - kv_helper is lmdb instance to access database
-        - process_wo_id is id of the work-order that is to be processed
+            @param process_wo_id - Id of the work-order that is to be processed
         """
-        logger.info("Processing work order")
-        try:
-            # Get all workorders requests from KV storage lookup and process
-            list_of_workorders = self._kv_helper.lookup("wo-scheduled")
-            if not list_of_workorders:
-                logger.info("Received empty list of work orders from " +
-                            "wo-scheduled table")
-                return
-        except Exception as e:
-            logger.error("Problem while getting ids from wo-scheduled table")
-            return
-
-        for wo_id in list_of_workorders:
-            if wo_id == process_wo_id:
-                resp = self._process_work_order_by_id(wo_id)
-                return resp
+        logger.info("Processing work orders found in wo-worker-pending table.")
+
+        wo_id = self._kv_helper.csv_pop("wo-worker-pending", self._worker_id)
```

Comment:
can we use `csv_peek` and `csv_pop` if  `process_wo_id == wo_id`

Reply:
Using csv_match_pop which does both this in a single remote db call.